app/javascript/components/advanced-settings-form/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import YAML from 'yaml';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './advanced-settings-form.schema';
+
+const AdvancedSettingsForm = ({ url }) => {
+  const [{ isLoading, initialValues }, setState] = useState({ isLoading: true });
+
+  useEffect(() => {
+    API.get(`${url}/settings`).then((settings) => setState({
+      initialValues: { // It is coming in as JSON but we need YAML
+        settings: YAML.stringify(settings),
+      },
+      isLoading: false,
+    }));
+  }, [url]);
+
+  const onSubmit = ({ settings }) => {
+    miqSparkleOn();
+    // The API expects us to send a JSON
+    API.patch(`${url}/settings`, YAML.parse(settings)).then(() => {
+      add_flash(__('Configuration changes saved'), 'success');
+      // Scroll to the top of the page to see the flash message
```

Comment:
why here?

If add_flash should scroll to the top, shouldn't it always?

Reply:
I don't know how to scroll it to the top :confused:

Reply:
You do, on the next line :D

The point was, should `add_to_flash` *always* scroll to the flash message?